### PR TITLE
Added yaml files for supporting Azure in Rooks deployments.

### PR DESCRIPTION
**Pull request by jgarcia-daishogroup**
#### What does this PR do?

This PR contains the modifications of the yaml files for supporting the deployment of Rook in a Kubernetes-based Azure cluster.

#### What are the relevant tickets?

- [NP-XXX](https://nalej.atlassian.net/browse/NP-XXX)

### CODE REVIEW

File: components/storage-fabric/appcluster/storage-fabric.rook.29.deployment.operator.yaml [hunk 1]
```diff
@@ -8,14 +8,13 @@
...
-# - operator-with-csi.yaml: Enable the ceph-csi driver
 #################################################################################################################
-
+# OLM: BEGIN OPERATOR DEPLOYMENT
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: rook-ceph-operator
-  namespace: __NPH_NAMESPACE
+  namespace: rook-ceph
```
Comment (dhiguero) [inline components/storage-fabric/appcluster/storage-fabric.rook.29.deployment.operator.yaml]: The previous namespace is the correct one as the compilation process will change it

```suggestion
  namespace: __NPH_NAMESPACE
```

File: components/storage-fabric/appcluster/storage-fabric.rook.30.cephcluster.yaml [hunk 2]
```diff
@@ -15,122 +15,80 @@ metadata:
   name: rook-ceph
   namespace: rook-ceph
```
Comment (dhiguero) [inline components/storage-fabric/appcluster/storage-fabric.rook.30.cephcluster.yaml]:
```suggestion
  namespace: __NPH_NAMESPACE
```

File: components/storage-fabric/appcluster/storage-fabric.rook.30.cephcluster.yaml [hunk 3]
```diff
@@ -15,122 +15,80 @@ metadata:
...
-  # enable the ceph dashboard for viewing cluster status
+    # A volume claim template can be specified in which case new monitors (and
+    # monitors created during fail over) will construct a PVC based on the
+    # template for the monitor's primary storage. Changes to the template do not
+    # affect existing monitors. Log data is stored on the HostPath under
+    # dataDirHostPath. If no storage requirement is specified, a default storage
+    # size appropriate for monitor data will be used.
+    volumeClaimTemplate:
+      spec:
+        storageClassName: managed-premium
```
Comment (dhiguero) [inline components/storage-fabric/appcluster/storage-fabric.rook.30.cephcluster.yaml]: As this file seems azure specific, it should be renamed as `storage-fabric.rook.30.cephcluster.yaml.azure` with the other version not being overwritten.

Comment (jgarcia-daishogroup) [reply components/storage-fabric/appcluster/storage-fabric.rook.30.cephcluster.yaml]: Added Azure and Baremetal setups.

**Comment by jgarcia-daishogroup** [comment]
Resolved issues raised by @dhiguero

**Pull request merged**